semantics/src/traversal/_utils.rs

```diff
@@ -0,0 +1,33 @@
+use crate::namespace::types::FixedSize;
+use crate::{
+    ExpressionAttributes,
+    Type,
+};
+use fe_parser::ast as fe;
+use fe_parser::span::{
+    Span,
+    Spanned,
+};
+
+/// Creates a new spanned expression. Useful in cases where an `Expr` is nested
+/// within the node of a `Spanned` object.
+pub fn spanned_expression<'a>(span: &Span, exp: &fe::Expr<'a>) -> Spanned<fe::Expr<'a>> {
+    Spanned {
+        node: (*exp).clone(),
+        span: (*span).to_owned(),
+    }
+}
+
+pub fn expression_attributes_to_types(attributes: Vec<ExpressionAttributes>) -> Vec<Type> {
+    attributes
+        .iter()
+        .map(|attributes| attributes.typ.clone())
+        .collect()
+}
+
+pub fn fixed_sizes_to_types(sizes: Vec<FixedSize>) -> Vec<Type> {
+    sizes
+        .iter()
+        .map(|param| param.clone().into_type())
+        .collect()
+}
```

Comment:
I'm not super happy with these two helper functions. I initially wanted to write an `impl` for `Vec<ExpressionAttributes>` but that isn't legal as I can't write an impl for a foreign type. There might be a more idiomatic way to handle that.

Reply:
We should be able to implement the relevant `From` traits and then use something like: `attributes.iter().collect<Type>()`, but this is ok as is.